facebook/OpenBIC
# fby3.5: cl: SB BIC supports LTC4282 sensor reading

fby3.5: cl: SB BIC supports LTC4282 sensor reading
Summary:
- SB BIC supports LTC4282 sensor device
  1. For the EVT3 stage, there are two different board revision IDs systems including EFUSE and HOTSWAP.
     A. For the EFUSE system, the HSC module type is MPS5990.
        The HSC sensors "HSC Temp," "HSC Input Vol," "HSC Output Cur" and "HSC Input Pwr" are read from HSC module.
     B. For the HOTSWAP system, there are three HSC types including ADM1272, LTC4282 and LTC4286.
        The HSC sensors "HSC Input Vol," "HSC Output Cur" and "HSC Input Pwr" are read from HSC module.
        BIC can judge the HSC module type according to the voltage of ADC-7.
        The HCS type is ADM1278 if the voltage is 0.5V(+-15%).
        The HCS type is LTC4282 if the voltage is 1.0V(+-15%).
        The HCS type is LTC4286 if the voltage is 1.5V(+-15%).
        For this revision system, the "HCS Temp" and "MB Outlet Temp" sensor values are read from TMP431 chip.
<TODO>
The GPIOA7 should be set as "H" if the 2OU is present.

Test Plan:
1. Build code: pass
2. Check HSC sensor values: pass

Test Log:
root@bmc-oob:~# sensor-util slot1|grep HSC
HSC Temp                     (0xE) :   35.44 C     | (ok)
HSC Input Vol                (0x29) :   12.48 Volts | (ok)
HSC Output Cur               (0x30) :    6.07 Amps  | (ok)
HSC Input Pwr                (0x39) :   77.20 Watts | (ok)
root@bmc-oob:~# power-util slot1 off
Powering fru 1 to OFF state...
root@bmc-oob:~# sensor-util slot1|grep HSC
HSC Temp                     (0xE) :   35.25 C     | (ok)
HSC Input Vol                (0x29) :   12.50 Volts | (ok)
HSC Output Cur               (0x30) :    0.07 Amps  | (ok)
HSC Input Pwr                (0x39) :    2.67 Watts | (ok)
